From 923ba2a1653b942093e9d25f8f4d2e7c09a66328 Mon Sep 17 00:00:00 2001
From: Giulio Benetti <giulio.benetti@benettiengineering.com>
Date: Wed, 8 Feb 2023 23:50:56 +0100
Subject: [PATCH] Remove atk-bridge support.

atk-bridge doesn't seem useful for now in Buildroot and requires to
add two new packages just for it: at-spi2-core and at-spi2-atk.

Signed-off-by: Hadrien Boutteville <hadrien.boutteville@gmail.com>
[Giulio: remove 'atk-bridge-2.0' too]
Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
---
 gtk/a11y/gtkaccessibility.c | 8 --------
 meson.build                 | 4 +---
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/gtk/a11y/gtkaccessibility.c b/gtk/a11y/gtkaccessibility.c
index 7f0e5207d9..14da1e80e2 100644
--- a/gtk/a11y/gtkaccessibility.c
+++ b/gtk/a11y/gtkaccessibility.c
@@ -37,10 +37,6 @@
 #include <gtk/gtktogglebutton.h>
 #include <gtk/gtkaccessible.h>
 
-#ifdef GDK_WINDOWING_X11
-#include <atk-bridge.h>
-#endif
-
 static gboolean gail_focus_watcher      (GSignalInvocationHint *ihint,
                                          guint                  n_param_values,
                                          const GValue          *param_values,
@@ -988,9 +984,5 @@ _gtk_accessibility_init (void)
   _gtk_accessibility_override_atk_util ();
   do_window_event_initialization ();
 
-#ifdef GDK_WINDOWING_X11
-  atk_bridge_adaptor_init (NULL, NULL);
-#endif
-
   atk_misc_instance = g_object_new (GTK_TYPE_MISC_IMPL, NULL);
 }
diff --git a/meson.build b/meson.build
index aed48fc3f6..370bf1cce1 100644
--- a/meson.build
+++ b/meson.build
@@ -564,7 +564,7 @@ if x11_enabled
   xfixes_dep     = dependency('xfixes', required: false)
   xcomposite_dep = dependency('xcomposite', required: false)
   fontconfig_dep = dependency('fontconfig', fallback: ['fontconfig', 'fontconfig_dep'])
-  atkbridge_dep  = dependency('atk-bridge-2.0', version: at_spi2_atk_req)
+  atkbridge_dep  = dependency('atk-bridge-2.0', version: at_spi2_atk_req, required: false)
 
   backend_immodules += ['xim']
 
@@ -583,8 +583,6 @@ if x11_enabled
     x11_pkgs += ['xdamage']
   endif
 
-  atk_pkgs += ['atk-bridge-2.0']
-
   cdata.set('HAVE_XDAMAGE', xdamage_dep.found() ? 1 : false)
   cdata.set('HAVE_XCURSOR', xcursor_dep.found() ? 1 : false)
   cdata.set('HAVE_XCOMPOSITE', xcomposite_dep.found() ? 1 : false)
-- 
2.34.1

